types/apk-definition.json

```diff
@@ -1,37 +1,59 @@
 {
   "$schema": "https://packageurl.org/schemas/purl-type-definition.schema-1.0.json",
-  "$id": "https://packageurl.org/types/bitbucket-definition.json",
+  "$id": "https://packageurl.org/types/apk-definition.json",
   "type": "apk",
   "type_name": "APK-based packages",
   "description": "Alpine Linux APK-based packages",
   "repository": {
     "use_repository": true,
-    "note": "There is no default package repository; this should be implied either from the distro qualifiers key  or using a repository base url as repository_url qualifiers key."
+    "note": "There is no default package repository; this should be implied either from the distro qualifiers key or using a repository base url as repository_url qualifiers key."
   },
   "namespace_definition": {
     "requirement": "required",
     "note": "The namespace is the vendor such as alpine or openwrt. It is not case sensitive and must be lowercased.",
     "native_name": "vendor",
-    "case_sensitive": false
+    "case_sensitive": false,
+    "normalization_rules": [
+      "It is not case sensitive and must be lowercased."
+    ]
   },
   "name_definition": {
+    "requirement": "required",
     "note": "The name is the package name. It is not case sensitive and must be lowercased.",
     "native_name": "name",
-    "case_sensitive": false
+    "case_sensitive": false,
+    "normalization_rules": [
+      "It is not case sensitive and must be lowercased."
+    ]
   },
   "version_definition": {
-    "note": "The version is a package version as expected by apk.",
+    "requirement": "required",
+    "note": "The version is a package version as expected by apk. The format is typically 'pkgver-rX', where 'pkgver' may include suffixes like '_alpha', '_p1', etc. The '-rX' release suffix is a strong convention but not strictly required for a version string to be valid.",
     "native_name": "version"
   },
   "qualifiers_definition": [
     {
       "key": "arch",
+      "requirement": "optional",
       "description": "The arch is the qualifiers key for a package architecture."
+    },
+    {
+      "key": "distro",
+      "requirement": "optional",
+      "description": "The distribution name when using multiple distributions"
+    },
```

Comment:
This doesn't make sense. I don't know if the problem is just that it's not defined properly. The distribution goes in the namespace. What does it mean if a PURL has a distribution in the namespace and in the qualifier? If it makes sense for there two be two distributions, what if there are more than two?